main/ejml-dsparse/src/org/ejml/sparse/csc/misc/ImplCommonOps_DSCC.java

```diff
@@ -38,6 +39,44 @@
  */
 public class ImplCommonOps_DSCC {
 
+    public static void select(DMatrixSparseCSC A, DMatrixSparseCSC output, IPredicateBinary selector) {
+        int selectCount = 0;
+
+        // size estimation
+        if (output != A) {
+            output.growMaxLength(A.nz_length / 2, false);
+        }
+
+        // selecting a subset doesn't change the order
+        output.indicesSorted = A.indicesSorted;
+
+        for (int col = 0; col < A.numCols; col++) {
+            int start = A.col_idx[col];
+            int end = A.col_idx[col+1];
+
+            output.col_idx[col] = selectCount;
+
+            if (output.nz_rows.length < (selectCount + (end - start))) {
+                int maxLength = Integer.max(output.nz_length * 2 + 1, A.nz_length);
+                output.growMaxLength(maxLength, true);
+            }
+
+            for (int i = start; i < end; i++) {
+                int row = A.nz_rows[i];
+
+                if(selector.apply(row, col)) {
```

Comment:
Suggestion: Format to avoid pyramiding.

```
for (int i = start; i < end; i++) {
    int row = A.nz_rows[i];

    if(!selector.apply(row, col))
        continue;

    output.nz_rows[selectCount] = row;
    output.nz_values[selectCount] = A.nz_values[i];
    selectCount++;
}
```

Reply:
with pyramiding you mean increasing indentation?

Reply:
```
 if(!selector.apply(row, col))
        continue;
```
I think this can be easily missed when reading the code. If you don't insist, i.e., its only a stylistic thing, I would like to keep my version.

Reply:
yep you can ignore it. Just a suggestion. It's in the coding standards at work and I've started to like it. When things become about 4 tabs deep then it becomes important.

https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)